enhancements/authentication/separate-oauth-resources.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Separate-OAuth-API-Resources
+authors:
+  - "@deads2k"
+reviewers:
+  - "@sttts"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-10-16
+last-updated: 2019-10-16
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Separate OAuth API-Resources
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+The API resources used by the OpenShift OAuth server will be moved to their own `openshift-oauth-apiserver` repository.
+The openshift-oauth-apiserver binary will be managed by the existing authentication-operator.
+
+## Motivation
+
+The OpenShift OAuth server requires several openshift API resources in order to function.
+Because the OAuth server enables access to the kube-apiserver, it needs to be able to run separately from the existing
+set of API resources that are focused on enabling a developer workflow.
+This narrowing of scope expands the deployment options for the OpenShift OAuth server.
+
+### Goals
+
+1. Make it possible to run the OpenShift OAuth server without installing all the developer focused API Resources.
+
+### Non-Goals
+
+1. Move routes out of the OpenShift API Server.
+
+## Proposal
+
+1. Create a new repo called `openshift-oauth-apiserver` that produces a binary called `openshift-oauth-apiserver`.
+2. Add the following API groups to the new repo
+   1. authorization  - most of this API group is deprecated.  It will sleep here until we're ready to retire it.
+   2. oauth - core oauth types
+   3. project - intersection of authorization and namespaces.  It is heavily dependent on the way our scopes work.
+   4. quota - intersection of users and namespaces
+   5. security - SCC
```

Comment:
@derekwaynecarr this is technically usable separately, but it cannot be written as a CRD (it's got a virtual projection). How many different units?